### PR TITLE
TestBasicAerQasmSimulator.test_qasm_simulator fails hourly

### PR DESCRIPTION
**Pull request by 1ucian0**
**tl;dr** : The test `test.python.basicaer.test_qasm_simulator.TestBasicAerQasmSimulator.test_qasm_simulator` seems to fail every now and then. Specifically, two o three times every hour. Exactly. The problem: we are not seeding the simulator.

**The long story**:
After noticing that, from time to time `test_qasm_simulator.TestBasicAerQasmSimulator.test_qasm_simulator` was failing in the CI, I the test 10k times in a sequence.
```sh
#!/bin/bash

TEST="test.python.basicaer.test_qasm_simulator.TestBasicAerQasmSimulator.test_qasm_simulator"

for run in {1..10000} ; do
   python -m unittest -q $TEST &> /dev/null  || echo "$(date +%Y%m%d:%H:%m:%S) : test failed"
done
```
The output:
```
20200318:23:03:29 : test failed
20200318:23:03:05 : test failed
20200319:00:03:35 : test failed
20200319:00:03:21 : test failed
20200319:01:03:02 : test failed
20200319:01:03:14 : test failed
20200319:01:03:16 : test failed
20200319:01:03:53 : test failed
20200319:01:03:57 : test failed
20200319:02:03:07 : test failed
20200319:02:03:44 : test failed
20200319:02:03:35 : test failed
20200319:02:03:33 : test failed
```

Noticed that test fails in baches separated by exactly one hour. I have no idea way. Maybe some sort of cycle in the numpy PRNG?

In any case, the problem was that we were not seeding the simulator properly.

I will try to find the seeds that created the failed cases.

### CODE REVIEW

**Comment by 1ucian0** [comment]
A failing seed: `1656390055`

```
diff --git a/test/python/basicaer/test_qasm_simulator.py b/test/python/basicaer/test_qasm_simulator.py
index 49958ef82..69cb22e0d 100644
--- a/test/python/basicaer/test_qasm_simulator.py
+++ b/test/python/basicaer/test_qasm_simulator.py
@@ -34,7 +34,7 @@ class TestBasicAerQasmSimulator(providers.BackendTestCase):
     def setUp(self):
         super(TestBasicAerQasmSimulator, self).setUp()

-        self.seed = 88
+        self.seed = 1656390055
         qasm_filename = self._get_resource_path('example.qasm', Path.QASMS)
         transpiled_circuit = QuantumCircuit.from_qasm_file(qasm_filename)
         transpiled_circuit.name = 'test'
```
```
09:56 $ python -m unittest  -v test.python.basicaer.test_qasm_simulator.TestBasicAerQasmSimulator.test_qasm_simulator
test_qasm_simulator (test.python.basicaer.test_qasm_simulator.TestBasicAerQasmSimulator)
Test data counts output for single circuit run against reference. ... FAIL

======================================================================
FAIL: test_qasm_simulator (test.python.basicaer.test_qasm_simulator.TestBasicAerQasmSimulator)
Test data counts output for single circuit run against reference.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "qiskit-terra/test/python/basicaer/test_qasm_simulator.py", line 131, in test_qasm_simulator
    self.assertDictAlmostEqual(counts, target, threshold)
AssertionError: ('010 010': 83 != 128.0) within 40.96 delta

----------------------------------------------------------------------
Ran 1 test in 0.339s

FAILED (failures=1)
```
